packages/ripple-binary-codec/test/tx-encode-decode.test.ts

```diff
@@ -116,4 +116,13 @@ describe('encoding and decoding tx_json', function () {
       encode(my_tx)
     }).toThrow()
   })
+
+  it('throws when there is an unknown field', function () {
+    const my_tx = Object.assign({}, tx_json, {
+      BadField: 1,
```

Comment:
If this field was titled `badField` (beginning with small letter), then the code does not throw. I understand the implementation of the code. But what is rationale for this behavior?

Reply:
SFields are always capitalized. Synthetic fields (non-serialized fields that are returned in RPC outputs) are lower-case.